### ISSUE TITLE
Lazy loading of hdus broke slicing of HDULists

### ISSUE THREAD

**Issue by nden** [astropy/astropy#5585]
I think #5065 broke operations like this:

```
f = fits.open('j94f05bgq_flt.fits')
[ext for ext in f[1:] if ext.header['EXTNAME'] == 'SCI']
[ext.header['CRVAL*'] for ext in f[1:] if ext.header['EXTNAME'] == 'SCI']
```

The file needs to be read in some way for this to work, e.g.
```
In [14]: f.info()
In [15]:[ext for ext in f[1:] if ext.header['EXTNAME'] == 'SCI']
Out[15]: 
[<astropy.io.fits.hdu.image.ImageHDU at 0x31f0f90>,
 <astropy.io.fits.hdu.image.ImageHDU at 0x3407890>]
```

**Comment by eteq** [comment]
@embray - do you understand what's going on here? That's a pretty serious incompatible change, and I suspect it isn't what was intended?

**Comment by mwcraig** [comment]
@nden -- I believe this has been fixed in #5589. Can you give it a try?
